tests/containers/workbenches/workbench_image_test.py

```diff
@@ -203,6 +202,8 @@ def grab_and_check_logs(subtests: pytest_subtests.SubTests, container: Workbench
         "JupyterEventsVersionWarning: The `version` property of an event schema must be a string. It has been type coerced, but in a future version of this library, it will fail to validate.",
         # This is a message from our reverse proxy nginx caused by the fact that we attempt to connect to the code-server before it's actually running (container.start(wait_for_readiness=True)).
         "connect() failed (111: Connection refused) while connecting to upstream, client",
+        # We use oauth-proxy to give us authentication, and we use OpenShift route to get HTTPS
+        "WARNING: The Jupyter server is listening on all IP addresses and not using encryption. This is not recommended.",
```

Comment:
This warning wasn't there before this change? :thinking: What particular change caused this to pop up now? I can see that we set 0.0.0.0 before, which should be quite same as what we do now :thinking:  or am I wrong?

Reply:
Okay, probably this change caused this to pop up?
```
    - ServerApp.ip=0.0.0.0 \
    + ServerApp.ip="" \
```

Reply:
yes, that one. 0.0.0.0 means all ipv4 interfaces, "" means all interfaces (ipv4 and 6) and causes the warning to pop up

Reply:
It's quite strange though... I see no difference WRT security between the "0.0.0.0" and ""...

Reply:
i do see logic of the message. Listening on 0.0.0.0 is not listening on "all IP addresses", only on ipv4 addresses, so the message does not appear then. Did not investigate further, and i think it does not deserve more investigation.

Reply:
it can be "all IP addresses" on IPv4 only stack; anyway, I believe this messages should be a security warning thus my point above; but yeah, no need to investigate further, that wasn't my intention.

Reply:
maybe @jesuino might want to take a look, if the conditions when the message is printed may need a bit of tuning...

Reply:
here's the code that produces the warning, I had to struggle a bit to find it

https://github.com/search?q=repo%3Ajupyter-server%2Fjupyter_server%20%22not%20using%20encryption%22&type=code

Reply:
Looking at
```
if not app.ip:
```

if we provided :: as the IP, the warning may go away...

Reply:
yeah, I was wondering why we don't have "::" there as it is used for the other image

anyway, I don't mind that much, just wanted to clarify this... we don't need to bother with this anyhow more, I suppose